gradle-module-plugin/src/main/kotlin/io/ia/sdk/gradle/modl/task/WriteModuleXml.kt

```diff
@@ -220,12 +215,32 @@ open class WriteModuleXml @Inject constructor(_objects: ObjectFactory) : Default
         return false
     }
 
-    private fun manifests(): List<ArtifactManifest> {
-
-        return artifactManifests.get().map { manifest ->
+    // Manifests' artifacts, collect + sort by distinct scopes to de-dup them
+    private fun manifests(): List<Pair<String,String>> =
+        artifactManifests.get().map { manifest ->
             artifactManifestFromJson(manifest.asFile.readText(Charsets.UTF_8))
+        }.let { manifests ->
+            manifests
+                .flatMap { mani -> mani.artifacts }
+                .groupBy { arti -> arti.jarName }
+                .map { (jar, artifacts) ->
+                    Pair(
+                        jar,
+                        // Combine all scopes for the artifact
+                        artifacts.fold(setOf<Char>()) { scope, arti ->
+                            scope.union(
+                                manifests
+                                    .filter { mani -> arti in mani.artifacts }
+                                    .flatMap { mani -> mani.scope.toList() }
+                            )
+                        }.joinToString("")
+                    )
+                }.sortedWith(
```

Comment:
The above is bulk of the refinement, minimizing the number of intermediate steps of the transform. It seems fucntionally equivalent to the original PRs given the unit test coverage. All of `WriteModuleXmlTest` passes.

Reply:
Also, all my listening to nerdy array language podcasts is finally paying off. I'll never touch APL, J, Julia, K, R, etc. but some of the mathematical underpinnings are landing in in the JVM APIs.